js/foundation.util.triggers.js

```diff
@@ -99,125 +150,132 @@ function closemeListener(pluginName) {
       return `closeme.zf.${name}`;
     }).join(' ');
 
-    $(window).off(listeners).on(listeners, function(e, pluginId){
-      let plugin = e.namespace.split('.')[0];
-      let plugins = $(`[data-${plugin}]`).not(`[data-yeti-box="${pluginId}"]`);
+    $(window).off(listeners).on(listeners, Triggers.Global.closeMeListener);
+  }
+}
 
-      plugins.each(function(){
-        let _this = $(this);
+function debounceGlobalListener(debounce, trigger, listener) {
+  let timer, args = Array.prototype.slice.call(arguments, 3);
+  $(window).off(trigger).on(trigger, function(e) {
+    if (timer) { clearTimeout(timer); }
+    timer = setTimeout(function(){
+      listener.apply(null, args);
+    }, debounce || 10);//default time to emit scroll event
+  });
+}
 
-        _this.triggerHandler('close.zf.trigger', [_this]);
-      });
-    });
+Triggers.Initializers.addResizeListener = function(debounce){
+  let $nodes = $('[data-resize]');
+  if($nodes.length){
+    debounceGlobalListener(debounce, 'resize.zf.trigger', Triggers.Listeners.Global.resizeListener, $nodes);
   }
 }
 
-function resizeListener(debounce){
-  let timer,
-      $nodes = $('[data-resize]');
+Triggers.Initializers.addScrollListener = function(debounce){
+  let $nodes = $('[data-scroll]');
   if($nodes.length){
-    $(window).off('resize.zf.trigger')
-    .on('resize.zf.trigger', function(e) {
-      if (timer) { clearTimeout(timer); }
-
-      timer = setTimeout(function(){
+    debounceGlobalListener(debounce, 'scroll.zf.trigger', Triggers.Listeners.Global.scrollListener, $nodes);
+  }
+}
 
-        if(!MutationObserver){//fallback for IE 9
-          $nodes.each(function(){
-            $(this).triggerHandler('resizeme.zf.trigger');
-          });
-        }
-        //trigger all listening elements and signal a resize event
-        $nodes.attr('data-events', "resize");
-      }, debounce || 10);//default time to emit resize event
+// TODO: Figure out if this needs to be refactored or removed
```

Comment:
@coreysyms can you help me understand what this function is doing? I'm confused because as currently utilized it appears to trigger a mutateme event on all `[data-mutate]` nodes whenever the global listeners are initialized (previously checkListeners)